artifact/sha/sha.hpp

```diff
@@ -58,6 +60,43 @@ extern const ErrorCategoryClass KeyValueParserErrorCategory;
 
 error::Error MakeError(ErrorCode code, const string &msg);
 
+class SHA {
+public:
+	SHA() :
+		sha_ {} {};
+	SHA(vector<uint8_t> &v) :
+		sha_ {v} {};
+	operator string() const {
+		std::stringstream ss {};
+		for (unsigned int i = 0; i < 32; ++i) {
+			ss << std::hex << std::setw(2) << std::setfill('0') << (int) sha_.at(i);
+		}
+		return ss.str();
+	}
+	operator vector<uint8_t>() const {
```

Comment:
For similarity with `std`, I would suggest the same here. Maybe `Data()`?

Reply:
Hmm, and then wrap `vector<uint8_t>.data()` you mean?

Reply:
Actually I was thinking

```
vector<uint8_t> Data() const
```

But maybe that's confusing since `vector.data()` already exists. Is `Vector()` a better name?

Reply:
Keep in mind that for example smart pointers, which are also doing this (behaving like pointers), they require you to call `get()` to get the view of the real, raw pointer. It's makes it explicit and searchable.

However, I won't oppose it, if you want to keep it as is. What you're saying makes sense, I just have a preference for being explicit (I think Golang reinforced that habit!).

Reply:
Noted, keeping it as it is for now though